source/proof_document/lib/proof_document.rb

```diff
@@ -0,0 +1,158 @@
+                   liveness_checking_enabled:,
+                   callback_url:)
+      @encryption_key = encryption_key
+      @front_image_iv = front_image_iv
+      @back_image_iv = back_image_iv
+      @selfie_image_iv = selfie_image_iv
+      @front_image_url = front_image_url
+      @back_image_url = back_image_url
+      @selfie_image_url = selfie_image_url
+      @liveness_checking_enabled = liveness_checking_enabled
+      @callback_url = callback_url
+    end
+
+    def proof(&callback_block)
+      set_up_env!
+
+      proofer_result = with_retries(**retry_options) do
+        document_proofer.post_images(
+          front_image: read_file(front_image_url, front_image_iv),
+          back_image: read_file(back_image_url, back_image_iv),
+          selfie_image: read_file(selfie_image_url, selfie_image_iv),
+          liveness_checking_enabled: liveness_checking_enabled,
+        )
+      end
+
+      result = proofer_result.to_h
+
+      result[:exception] = proofer_result.exception.inspect if proofer_result.exception
+
+      callback_body = {
+        document_result: result,
+      }
+
+      if block_given?
+        yield callback_body
+      else
+        post_callback(callback_body: callback_body)
+      end
+    end
+
+    def post_callback(callback_body:)
+      with_retries(**retry_options) do
+        Faraday.post(
+          callback_url,
+          callback_body.to_json,
+          "X-API-AUTH-TOKEN" => api_auth_token,
+          "Content-Type" => 'application/json',
+          "Accept" => 'application/json'
+        )
+      end
+    end
+
+    def api_auth_token
+      @api_auth_token ||= ENV.fetch("IDP_API_AUTH_TOKEN") do
+        ssm_helper.load('document_proof_result_lambda_token')
+      end
+    end
+
+    def set_up_env!
+      %w[
+        acuant_assure_id_password
+        acuant_assure_id_subscription_id
+        acuant_assure_id_url
+        acuant_assure_id_username
+        acuant_facial_match_url
+        acuant_passlive_url
+        acuant_timeout
+      ].each do |env_key|
+        ENV[env_key] ||= ssm_helper.load(env_key)
+      end
+    end
+
+    def document_proofer
+      IdentityDocAuth::Acuant::AcuantClient.new(
+        assure_id_password: ENV['acuant_assure_id_password'],
+        assure_id_subscription_id: ENV['acuant_assure_id_subscription_id'],
+        assure_id_url: ENV['acuant_assure_id_url'],
+        assure_id_username: ENV['acuant_assure_id_username'],
+        facial_match_url: ENV['acuant_facial_match_url'],
+        passlive_url: ENV['acuant_passlive_url'],
+        timeout: ENV['acuant_timeout'],
+        # exception_notifier: method(:notify_exception),
+      )
+    end
+
+    def ssm_helper
+      @ssm_helper ||= SsmHelper.new
+    end
+
+    def retry_options
+      {
+        max_tries: 3,
+        rescue: [Faraday::TimeoutError, Faraday::ConnectionFailed],
+      }
+    end
+
+    private
+
+    def ec2_data
+      @ec2_data ||= LoginGov::Hostdata::EC2.load
+    end
```

Comment:
The infrastructure that the `identity-hostdata` gem relies on...I don't think it will work inside the lambda....?

cc @crissupb can you confirm that the `http://169.254.169.254/2016-09-02/dynamic/instance-identity/document` URL will or will not work on a lambda?

Reply:
You are right.  No instance metadata.  Good catch!